circuits/cpp/src/aztec3/circuits/abis/state_read.hpp

```diff
@@ -64,6 +65,8 @@ template <typename NCT> struct StateRead {
         storage_slot.set_public();
         current_value.set_public();
     }
+
+    boolean is_empty() const { return storage_slot == 0; }
```

Comment:
I'm wondering if this is safe enough (and I don't know the answer).
It makes sense to only check the `storage_slot` and ignore the other data members in the context of how it's been called in the rest of this PR.
I just wonder if, in future, someone might misinterpret this function (based on its name) to mean "is every element of this struct default-initialised".?
Open question